src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/TimelineEntity.kt

```diff
@@ -0,0 +1,84 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity
+
+import jakarta.persistence.CascadeType
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.OneToMany
+import jakarta.persistence.OrderBy
+import jakarta.persistence.Table
+import jakarta.validation.constraints.NotNull
+import org.hibernate.Hibernate
+import org.hibernate.annotations.CreationTimestamp
+import org.hibernate.annotations.UpdateTimestamp
+import org.hibernate.annotations.UuidGenerator
+import java.time.Instant
+import java.util.UUID
+
+@Table(name = "timeline")
+@Entity
+class TimelineEntity(
+  @Id
+  @GeneratedValue
+  @UuidGenerator
+  var id: UUID? = null,
+
+  @Column(updatable = false)
+  @field:NotNull
+  var reference: UUID? = null,
+
+  @Column(updatable = false)
+  @field:NotNull
+  var prisonNumber: String,
+
+  @OneToMany(cascade = [CascadeType.ALL], orphanRemoval = true, fetch = FetchType.LAZY)
+  @JoinColumn(name = "timeline_id", nullable = false)
+  @OrderBy(value = "createdAt")
+  @field:NotNull
+  var events: MutableList<TimelineEventEntity>? = null,
+
+  @Column(updatable = false)
+  @CreationTimestamp
```

Comment:
Do these annotations work? (created/updated timestamps)
In the other entity classes we needed to annotate the class with `@EntityListeners(AuditingEntityListener::class)` - but that might have been to get the spring managed fields populated (created/updated by)

Reply:
Yes they are populated, but you've highlighted the need for an assertion for this in the ITs, which I've just added 👍